torch/multiprocessing/reductions.py

```diff
@@ -127,6 +130,10 @@ def reduce_storage(storage):
         metadata = (df, size)
         rebuild = rebuild_storage_fd
 
+    # WARNING!  This call to _weak_ref could lead to O(n) deleter
+    # behavior, if you repeatedly call it on the same Storage (all
+    # other sites are guarded by shared_cache; maybe this site
+    # should be too?)
```

Comment:
This warning applies to multiple places in this file I think

Reply:
Actually, it doesn't, because in all the other cases, there's a test if it's already in `shared_cache`, as the comment says :)